### ISSUE TITLE
RSPEC-5247: Disabling auto-escaping features is security-sensitive

### ISSUE THREAD

**Issue by eric-therond-sonarsource** [SonarSource/SonarJS#1947]
Implements [RSPEC-5247](https://jira.sonarsource.com/browse/RSPEC-5247)

For both JS/TS

Code examples: https://github.com/SonarSource/security-expected-issues/tree/master/javascript/rules/hotspots/RSPEC-5247

### Mustache
Report on `Mustache.escape` when a user redefines the escape function to the behaviour returning the only parameter without changes:
```
const Mustache = require("mustache");
Mustache.escape = function(text) {return text;}; // Sensitive
// or
Mustache.escape = a => a; // Sensitive
```

### kramed
Report on the property `sanitize` when set to `false`:
```js
let kramed = require('kramed');

var options = {
  renderer: new kramed.Renderer({
    sanitize: false // Sensitive
  })
};
``` 

For all other libraries check RSPEC as pattern to report is straightforward. Report on property disabling the auto-escape (e.g. `noEscape: true`) as primary location. Secondary location should be on method callee using these options
```js
const Handlebars = require('handlebars');
let source = "<p>attack {{name}}</p>";
let template = Handlebars.compile(source, { noEscape: true }); // Sensitive
    secondary   ^^^^^^^^^^^^^^^^^^^^                  ^^^^^^^^^^^^^^^ primary
```

**Comment by eric-therond-sonarsource** [comment]
Reopened because of [one FN](https://peach.sonarsource.com/code?id=security-expected-issues_nodev8&selected=security-expected-issues_nodev8%3Ajavascript%2Frules%2Fhotspots%2FRSPEC-5247%2Fmustache%2Fnoncompliant.js) for the mustache module:
```
  var template = document.getElementById('template').innerHTML;
  Mustache.escape = function(text) {return text;}; // FN
  var rendered = Mustache.render(template, { name: inputName });
  document.getElementById('target').innerHTML = rendered;
```

**Comment by saberduck** [comment]
I am closing the issue, so it's included in 6.7 milestone. We can create another issue for eventual FN
